extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -63,7 +65,28 @@ public void validate(ValidationContext validationContext) {
                             // No need to validate properties with default values
                             continue;
                         }
-                        String propertyName = configProperty.value("name").asString();
+                        AnnotationValue nameValue = configProperty.value("name");
+                        String propertyName;
+                        if (nameValue != null) {
+                            propertyName = nameValue.asString();
+                        } else {
+                            // org.acme.Foo.config
+                            if (injectionPoint.isField()) {
+                                FieldInfo field = injectionPoint.getTarget().asField();
+                                propertyName = (field.declaringClass().enclosingClass() == null ? field.declaringClass().name()
+                                        : field.declaringClass().enclosingClass() + "." + field.declaringClass().simpleName())
+                                        + "." + field.name();
```

Comment:
I know this is nitpicking but I'd probably turn the dot `.` into a a constant you then reuse. E.g. `final string DOT = "."`

Reply:
A constant is probably not necessary but I'll extract the logic into a shared util method.